indexing-service/src/main/java/io/druid/indexing/overlord/HeapMemoryTaskStorage.java

```diff
@@ -198,6 +197,30 @@ public void addLock(final String taskid, final TaskLock taskLock)
   }
 
   @Override
+  public void upgradeLock(String taskid, TaskLock taskLock)
+  {
+    giant.lock();
+
+    try {
+      Preconditions.checkNotNull(taskid, "taskId");
+      Preconditions.checkNotNull(taskLock, "taskLock");
+      TaskLock taskLockToRemove = taskLock.withPreemptive(!taskLock.isPreemptive());
+      if(!taskLocks.remove(taskid, taskLockToRemove)) {
+        log.error("No lock found on interval [%s] and version [%s] for task: [%s] to be upgraded to non-preemptive lock ! This is bad !",
+                  taskLock.getInterval(),
+                  taskLock.getVersion(),
+                  taskid
+        );
```

Comment:
this should be an exception, right? I would think that since this method is `void` it should only return if the upgraded lock actually does exist. otherwise you have no real way of knowing if it worked.